### ISSUE TITLE
PDFLoader error running ingest 

### ISSUE THREAD

**Issue by hillis** [hwchase17/langchainjs#699]
Running a PDFLoader to ingest data to Vectorstore and get the follow error.

message: 'The browser/environment lacks native support for critical functionality used by the PDF.js library (e.g. `Path2D` and/or `ReadableStream`); please use a `legacy`-build instead.',
  name: 'UnknownErrorException',
  details: 'Error: The browser/environment lacks native support for critical functionality used by the PDF.js library (e.g. `Path2D` and/or `ReadableStream`); please use a `legacy`-build instead.'

Workaround was going into your node_module/dist/document_loaders/fs/pdf.js.  Scroll to the bottom and change import pdfjs-dist to const { default: mod } = await import("pdfjs-dist/legacy/build/pdf.js");

**Comment by nfcampos** [comment]
Yes this is as expected. Note the legacy build installs polyfills into your environment so we really cannot make it the default sadly.

**Comment by okomarov** [comment]
Maybe allow `pdf-parse` or default to that one? Would you accept a PR with a switch for which dependency to use?

**Comment by nfcampos** [comment]
We have this in the docs now https://js.langchain.com/docs/modules/indexes/document_loaders/examples/file_loaders/pdf#usage-legacy-environments anything you'd change/improve?

**Comment by okomarov** [comment]
> We have this in the docs now https://js.langchain.com/docs/modules/indexes/document_loaders/examples/file_loaders/pdf#usage-legacy-environments anything you'd change/improve?

Thanks for the example, it really shows how easy it's to change the pdf importer to legacy. 

However, I don't quite understand what **legacy** means in this context. We're running node v19 and that surely is not legacy but it does require the legacy distribution. I think the distinction here is whether pdfjs is run in a browser vs a node environment?

I would actually default to the legacy distribution and rename/rework that paragraph of the docs to show how to change import, given that:
1. ML pipelines usually live in the backend
2. You are actually listing a Node.js tag in your docs:
<img width="876" alt="image" src="https://user-images.githubusercontent.com/3731173/232324952-44387987-9d15-4edf-99be-1f76eccd37e6.png">

**Comment by gpeal** [comment]
+1 to @okomarov. I've been pretty confused by this while running langchain in a modern node environment

**Comment by okomarov** [comment]
@nfcampos Thanks for getting this through so quickly, really appreciated!

**Comment by nfcampos** [comment]
This has been merged
